op-service/txmgr/metrics/tx_metrics.go

```diff
@@ -42,10 +63,59 @@ func MakeTxMetrics(ns string, factory metrics.Factory) TxMetrics {
 			Help:      "Latency of a confirmed transaction in milliseconds",
 			Subsystem: "txmgr",
 		}),
+		currentNonce: factory.NewGauge(prometheus.GaugeOpts{
+			Namespace: ns,
+			Name:      "current_nonce",
+			Help:      "Current nonce of the from address",
+			Subsystem: "txmgr",
+		}),
+		txConfirmed: factory.NewCounterVec(prometheus.CounterOpts{
+			Namespace: ns,
+			Name:      "tx_confirmed_count",
+			Help:      "Count of the transactions that have been confirmed. Label is receipt status.",
+			Subsystem: "txmgr",
+		}, []string{"status"}),
+		txPublished: factory.NewCounter(prometheus.CounterOpts{
+			Namespace: ns,
+			Name:      "tx_published_count",
+			Help:      "Count when transactions are submitted",
+			Subsystem: "txmgr",
+		}),
+		txPublishError: factory.NewCounterVec(prometheus.CounterOpts{
+			Namespace: ns,
+			Name:      "tx_publish_error_count",
+			Help:      "Count of publish errors. Labells are sanitized error strings",
+			Subsystem: "txmgr",
+		}, []string{"error"}),
+		lastPublishTime: factory.NewGauge(prometheus.GaugeOpts{
+			Namespace: ns,
+			Name:      "last_publish_time_unix",
+			Help:      "Unix timestamp of the last tx publish",
+			Subsystem: "txmgr",
+		}),
+		lastConfirmTime: factory.NewGauge(prometheus.GaugeOpts{
+			Namespace: ns,
+			Name:      "last_confirm_time_unix",
+			Help:      "Unix timestamp of the last tx confirmation",
+			Subsystem: "txmgr",
+		}),
```

Comment:
Just realize that this is a repetition of our `op-service/metrics/Event{,Vec}` implementations. You can reuse them to remove quite some code.